pkg/globalnet/controllers/global_egressip_controller.go

```diff
@@ -291,6 +291,12 @@ func (c *globalEgressIPController) onDelete(numRequeues int, globalEgressIP *sub
 
 	namedIPSet := c.newNamedIPSet(key)
 
+	if len(globalEgressIP.Status.AllocatedIPs) == 0 && len(podWatcher.AllocatedIPs) > 0 {
+		// Refer to issue for more details: https://github.com/submariner-io/submariner/issues/2388
```

Comment:
Wouldn’t it be better to requeue instead, so the delete will be retried later (once the creation has been processed and the cache updated)?

Reply:
That is what the existing code was doing when it failed to delete the IPset (as its referenced in the stale IPtable rule). I noticed that even as part of the delete re-queue event the globalEgressIPs.Status.AllocatedIPs field was empty.

Reply:
It feels like we’re papering over the issue — what happens with this fix if you create two `GlobalEgressIP`s and delete them in quick succession?

Reply:
Ah right, that’s the information I was missing. Thanks!